src/arrayMethodJoin.js

```diff
@@ -3,9 +3,29 @@
 /**
  * Implement method join
  */
+
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    let value = separator;
+
+    if (value === undefined) {
+      value = ',';
+    }
```

Comment:
you can simplify it by adding the default value on line 8 (`separator = ','`) or:
```suggestion
    const value = separator || ',';
```